src/World.cpp

```diff
@@ -1423,6 +1425,17 @@ void cWorld::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_Blo
 		}
 	}
 
+	float ParticleFormula = static_cast<float>(a_ExplosionSize) * 0.33f;
+	float Spread = ParticleFormula * 0.5f;
+	float ParticleCount = std::min((ParticleFormula * 125), 600.0f);
+
+	BroadcastParticleEffect("largesmoke", Vector3f(a_BlockX, a_BlockY - 0.5f, a_BlockZ), Vector3f{}, Spread, ParticleCount);
```

Comment:
Pull the `Vector3f(a_BlockX, a_BlockY - 0.5f, a_BlockZ)` out into a temporary variable, use it here and on line 1437. You'll need to static_cast the coords to `float`, or maybe make it a `Vector3d`, it might go through a silent conversion.